platform/platform-impl/src/com/intellij/openapi/wm/impl/ToolWindowsPane.java

```diff
@@ -1127,14 +1127,29 @@ public final void run() {
           myLayeredPane.validate();
           myLayeredPane.repaint();
         }
-        transferFocus();
+        transferExistingFocus(myInfo.getId());
       }
       finally {
         finish();
       }
     }
   }
 
+  private void transferExistingFocus(@Nullable String id) {
+    if (id == null) {
+      return;
+    }
+    ToolWindow toolWindow = myManager.getToolWindow(id);
+    if (toolWindow == null) {
+      return;
+    }
+    Component toolComponent = toolWindow.getComponent();
+    Component owner = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
```

Comment:
I think this should use `IdeFocusManager.getFocusOwner()` instead. Can you please make this change and check that the code still works as you need?

Reply:
Yes, using the IdeFocusManager works fine.
I updated the pull request.